ext/objspace/objspace_dump.c

```diff
@@ -416,7 +416,7 @@ dump_object(VALUE obj, struct dump_config *dc)
 
     size_t shape_id = rb_obj_shape_id(obj);
     dump_append(dc, ", \"shape_id\":");
-    dump_append_sizet(dc, shape_id);
+    dump_append_sizet(dc, shape_id >> SHAPE_ID_SHIFT);
```

Comment:
One awkward thing I did envision is that now we could dump a `shape_id` that isn't listed in the shapes.

So I decided to "untag" the shape IDs for now.